backend/frontend/tests/test_restricted_user_board_access.py

```diff
@@ -0,0 +1,159 @@
+from guardian.shortcuts import assign_perm
+from core.models import Team
+from core.tests.factories import UserFactory, TeamFactory
+from .frontend_test_mixin import FrontendTestMixin
+
+
+class TestUserBoardControlledAccess(FrontendTestMixin):
```

Comment:
A better name would be something like TestUserBoard Permissions or ...Authorization.

Controllers are a whole thing.